storage/oasis/client.go

```diff
@@ -4,107 +4,79 @@ package oasis
-	// TODO: Remove the hardcoded values in next block once indexer config supports multiple nodes.
-	// The plan is to introduce a new implementation of ConsensusApiLite that
-	// will forward each call to either CobaltConsensusApiLite or DamaskConsensusApiLite,
-	// depending on the `height` parameter in the call.
-	var nodeApi nodeapi.ConsensusApiLite
-	if cf.network.ChainContext == "53852332637bacb61b91b6411ab4095168ba02a50be4c3f82448438826f23898" {
-		// Cobalt mainnet.
-		creds := credentials.NewTLS(&tls.Config{MinVersion: tls.VersionTLS12})
-		dialOpts := []grpc.DialOption{grpc.WithTransportCredentials(creds)}
-		grpcConn, err := cmnGrpc.Dial(cf.network.RPC, dialOpts...)
+	var sdkConn connection.Connection
+	if sourceConfig.FastStartup {
+		var err error
+		sdkConn, err = connection.ConnectNoVerify(ctx, fakeNet)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("SDK ConnectNoVerify: %w", err)
 		}
-		nodeApi = cobalt.NewCobaltConsensusApiLite(grpcConn)
 	} else {
-		// Assume Damask.
-		client := (*cf.connection).Consensus()
-		nodeApi = damask.NewDamaskConsensusApiLite(client)
-	}
-
-	// Configure chain context for all signatures using chain domain separation.
-	signature.SetChainContext(cf.network.ChainContext)
-
-	c := &ConsensusClient{
-		nodeApi: nodeApi,
-		network: cf.network,
+		var err error
+		sdkConn, err = connection.Connect(ctx, fakeNet)
+		if err != nil {
+			return nil, fmt.Errorf("SDK Connect: %w", err)
+		}
 	}
-
-	return c, nil
+	return sdkConn, nil
 }
 
-// Runtime creates a new RuntimeClient.
-func (cf *ClientFactory) Runtime(runtimeID string) (*RuntimeClient, error) {
-	ctx := context.Background()
+func NewConsensusClient(ctx context.Context, sourceConfig *config.SourceConfig) (*ConsensusClient, error) {
+	return &ConsensusClient{
+		nodeApi: &history.HistoryConsensusApiLite{
+			History: history.Networks[sourceConfig.ChainName],
+		},
+		network: sdkConfig.DefaultNetworks.All[sourceConfig.ChainName],
+	}, nil
+}
 
-	connection := *cf.connection
-	client := connection.Runtime(&config.ParaTime{
-		ID: runtimeID,
-	})
+func NewRuntimeClient(ctx context.Context, sourceConfig *config.SourceConfig, runtime analyzer.Runtime) (*RuntimeClient, error) {
+	sdkConnect, err := SDKConnect(ctx, sourceConfig)
+	if err != nil {
+		return nil, err
+	}
+	fakePT := &sdkConfig.ParaTime{
+		ID: runtime.String(),
+	}
+	client := sdkConnect.Runtime(fakePT)
 
+	sdkNet := sdkConfig.DefaultNetworks.All[sourceConfig.ChainName]
 	// Configure chain context for all signatures using chain domain separation.
-	signature.SetChainContext(cf.network.ChainContext)
+	signature.SetChainContext(sdkNet.ChainContext)
 
 	info, err := client.GetInfo(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	rtCtx := runtimeSignature.DeriveChainContext(info.ID, cf.network.ChainContext)
+	rtCtx := sdkSignature.DeriveChainContext(info.ID, sdkNet.ChainContext)
 	return &RuntimeClient{
 		client:  client,
-		network: cf.network,
+		network: sdkNet,
```

Comment:
how did this used to work? wasn't cf.network the fake one we used with empty .ParaTimes?

Reply:
oh nvm we rummage around in sdkConfig.DefaultNetworks.All[?].ParaTimes.All[?]